changelog/_unreleased/2025-06-12-added-primary-order-delivery-and-transaction-to-cart.md

```diff
@@ -0,0 +1,15 @@
+---
+title: Added primary order delivery and transaction to cart
+---
+# Core
+* Changed `Checkout/Cart/Order/OrderConverter.php` to use the primary delivery id in the `convertDeliveries` and added the new cart extensions `ORIGINAL_PRIMARY_ORDER_TRANSACTION` and `ORIGINAL_PRIMARY_ORDER_DELIVERY`
+* Changed `Checkout/Cart/Delivery/DeliveryProcessor::process` to use the primary order delivery id
+___
+# Upgrade Information
+## Use of the primary order delivery
+### The methods call `$deliveries->first()` in the `Checkout/Cart/Order/OrderConverter.php` and `Checkout/Cart/Delivery/DeliveryProcessor.php` are replaced by the primary order delivery  
+## Delivery address sorting in OrderConverter
+### From now on, the first delivery address is always the primary order delivery to ensure that all remaining calls such as `$deliveries->first()` return the primary order delivery
+## Added two more cart extensions in the OrderConverter 
+### First, the new extension is `ORIGINAL_PRIMARY_ORDER_DELIVERY` which will return the primary order delivery
+### The second new extension is called `ORIGINAL_PRIMARY_ORDER_TRANSACTION` and will return the primary order transaction
```

Comment:
Please replace the headings with text stating:
- OrderConverter moves primary delivery to be the first, supporting legacy behaviour with `deliveries->first()`
- Primary ID extensions are added to the cart, but (!) changing them do not change the primary delivery / transaction

And as title, something along the line "primary order and transaction deliveries during recalculation"